File: eps/src/frontend/src/services/visitor-review.service.js
import axios from 'axios';

const API_URL = '/api/v1/visitor-review';

class VisitorReviewService {
    submit(visitorReview) {
        return axios
            .post(API_URL + '/submit', {
                username: visitorReview.username,
                studentId: visitorReview.studentId,
                email: visitorReview.email,
                department: visitorReview.department,

                visitorName: visitorReview.visitorName,
                idNumber: visitorReview.idNumber,
                phoneNumber: visitorReview.phoneNumber,
                campus: visitorReview.campus,
                gate: visitorReview.gate,
                startDate: visitorReview.startDate,
                endDate: visitorReview.endDate,

                submitDate: visitorReview.submitDate,
                status: 0
            })
    }
    getAll() {
        return axios
            .get(API_URL)
            .then(response => {
                return response;
            })
    }
    getAllByStudentId(studentId) {
        return axios
            .get(API_URL + "/" +studentId, {
                studentId: studentId
            })
            .then(response => {
                return response;
            })
    }
    changeStatus(visitorReview, status) {
        return axios
            .put(API_URL + "/" + visitorReview.studentId + "?submitDate=" + visitorReview.submitDate.replace(" ", "T") + "&status=" + status)
            .then(response => {
                return response;
            })
    }
}

export default new VisitorReviewService();

File: eps/src/main/java/com/company/visitorreview/VisitorReviewService.java
package com.company.visitorreview;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.time.LocalDateTime;
import java.util.List;

@Service
public class VisitorReviewService {

    private final VisitorReviewRepository visitorReviewRepository;

    @Autowired
    public VisitorReviewService(VisitorReviewRepository visitorReviewRepository) {
        this.visitorReviewRepository = visitorReviewRepository;
    }

    public List<VisitorReview> getVisitorApplications() { return visitorReviewRepository.findAll(); }

    public List<VisitorReview> getVisitorApplicationsByStudentId(Integer studentId) {
        return visitorReviewRepository.findAllByStudentId(studentId);
    }

    public void addNewVisitorApplication(VisitorReview visitorReview) {
        var newVisitorApplication = VisitorReview.builder()
                .username(visitorReview.getUsername())
                .studentId(visitorReview.getStudentId())
                .email(visitorReview.getEmail())
                .department(visitorReview.getDepartment())
                .visitorName(visitorReview.getVisitorName())
                .idNumber(visitorReview.getIdNumber())
                .phoneNumber(visitorReview.getIdNumber())
                .campus(visitorReview.getCampus())
                .gate(visitorReview.getGate())
                .startDate(visitorReview.getStartDate())
                .endDate(visitorReview.getEndDate())
                .submitDate(visitorReview.getSubmitDate())
                .status(0)
                .build();

        visitorReviewRepository.save(newVisitorApplication);
    }

    public void changeStatus(Integer studentId, LocalDateTime submitDate, Integer status) {
        List<VisitorReview> visitorReviewList = visitorReviewRepository.findAllByStudentId(studentId);
        for (VisitorReview visitorReview : visitorReviewList) {
            if (visitorReview.getSubmitDate().toString().equals(submitDate.toString())) {
                visitorReview.setStatus(status);
                visitorReviewRepository.save(visitorReview);
                break;
            }
        }
    }
}
